### ISSUE TITLE
could use some documentation on practical deployment of this driver

### ISSUE THREAD

**Issue by bkcsfi** [vmware/vsphere-flocker-driver#17]
Hi,

I'm considering using this driver on my small vSphere cluster but have some questions about the practical nature of it's use.

It seems a volume (hard disk) stored in the vsphereDatastore may move from host to host.

1. These volumes (hard disks) can only be attached to one host at a time, right?

2. When the volumes are attached to a host, are they attached as dependent or independent disks?

3. assuming they are attached as persistent, right?

4. How am I supposed to back up these volumes using something like Veeam? Since I don't know which host the volume will be attached to at any given time, I'd have to backup all the docker hosts. If these are independent disks, I don't think Veeam will back them up.

thanks

**Comment by pratikvgupta** [comment]
1.   These volumes (hard disks) can only be attached to one host at a time, right?
[Pratik] Yes. This is taken care automatically when you want to change the container host.
It will be detached from one container host and attached to another.

2.    When the volumes are attached to a host, are they attached as dependent or independent disks?
[Pratik] They are attached as dependent.

3.    assuming they are attached as persistent, right?
[Pratik] Yes they are persistent.

4.    How am I supposed to back up these volumes using something like Veeam? Since I don't know which host the volume will be attached to at any given time, I'd have to backup all the docker hosts. If these are independent disks, I don't think Veeam will back them up.
[Pratik] These are not independent so it should work.

Please let me know if you have more questions regarding this.

**Comment by moander** [comment]
Some security pointers would be nice. I don't sleep well knowing that the flocker agent has full access to my vCenter environment ;)

**Comment by jm66** [comment]
Hi @moander 

Narrowing down vCenter permissions, I think (haven't tested yet) the flocker driver would work with the following (please, @pratikvgupta, correct me if I'm mistaken):

- Datastore.AllocateSpace
- Datastore.Browse
- Datastore.Config
- Datastore.DeleteFile
- Datastore.FileManagement
- Datastore.Move
- Datastore.Rename
- Datastore.UpdateVirtualMachineFiles
- Datastore.UpdateVirtualMachineMetadata
- StorageProfile.View
- StorageViews.View
- System.Anonymous
- System.Read
- System.View
- Task.Create
- Task.Update
- VirtualMachine.Config.AddExistingDisk
- VirtualMachine.Config.AddNewDisk
- VirtualMachine.Config.AddRemoveDevice
- VirtualMachine.Config.AdvancedConfig
- VirtualMachine.Config.Annotation
- VirtualMachine.Config.DiskExtend
- VirtualMachine.Config.DiskLease
- VirtualMachine.Config.EditDevice
- VirtualMachine.Config.RemoveDisk
- VirtualMachine.Provisioning.DiskRandomAccess
- VirtualMachine.Provisioning.DiskRandomRead
- VirtualMachine.Provisioning.GetVmFiles
- VirtualMachine.Provisioning.PromoteDisks
- VirtualMachine.Provisioning.PutVmFiles-

**Comment by pratikvgupta** [comment]
Yes we need the above permissions for invoking the different VC APIs we use internally in the driver.

**Comment by jheller** [comment]
It would be helpful if this information was put in the README, or in  wiki page. The lack of documentation has caused a lot of frustration here.
